engine/src/test/java/org/camunda/bpm/engine/test/history/HistoricTaskInstanceTest.java

```diff
@@ -1132,4 +1132,28 @@ public void testQueryByProcessInstanceBusinessKeyAndArray() {
     assertEquals(0, query.processInstanceBusinessKeyIn(businessKey1, businessKey2).processInstanceBusinessKey(unexistingBusinessKey).count());
     assertEquals(1, query.processInstanceBusinessKeyIn(businessKey2, businessKey3).processInstanceBusinessKey(businessKey2).count());
   }
+
+  @Deployment(resources = { "org/camunda/bpm/engine/test/api/history/multiprocess/rootProcess.bpmn",
+      "org/camunda/bpm/engine/test/api/history/multiprocess/secondLevelProcess.bpmn",
+      "org/camunda/bpm/engine/test/api/history/multiprocess/thirdLevelProcess.bpmn" })
+  @Test
+  public void testQueryByRootProcessInstanceId() {
+    // given process definitions
+
+    // when
+    String rootProcessId = runtimeService.startProcessInstanceByKey("root-process").getId();
+    runtimeService.startProcessInstanceByKey("process-3");
+    while (historyService.createHistoricProcessInstanceQuery().unfinished().count() > 0) {
+      taskService.createTaskQuery().list().forEach(task -> taskService.complete(task.getId()));
+    }
```

Comment:
🔧 I feel like these should be in the `// given` section and the `// when` should be the query call with the rootProcessInstanceId (`.createHistoricTaskInstanceQuery().rootProcessInstanceId()`) because that is what this test is about.
What do you think?

Reply:
Yes indeed, that makes sense. Fixed!